runtime/expr/function/flatten.go

```diff
@@ -11,15 +12,18 @@ import (
 // https://github.com/brimdata/zed/blob/main/docs/language/functions.md#flatten
 type Flatten struct {
 	zcode.Builder
-	keyType zed.Type
-	mapper  *zed.Mapper
-	zctx    *zed.Context
+	keyType    zed.Type
+	mapper     *zed.Mapper
+	entryTypes map[int]zed.Type
```

Comment:
Nit: This obviously works either way, but I think the intention is clearer if it's a map from zed.Type to zed.Type.
```suggestion
	entryTypes map[zed.Type]zed.Type
```

Reply:
Good point.